#!/usr/bin/env python
# Motion configuration

PACKAGE = 'pandora_vision_motion'


from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0, "An In.....", 50, 0, 100)
# name type level description def min max

# ---------------------------- Threshold parameters ---------------------------
gen.add("diff_threshold", int_t, 0,
        "The background-foreground difference threshold", 45, 0, 7500)
gen.add("high_motion_threshold", double_t, 0, "", 7500, 0, 10000)
gen.add("low_motion_threshold", double_t, 0, "", 200, 0, 2000)
gen.add("max_standard_deviation", double_t, 0, "", 25, 0, 200)


# -------------------------- Visualization parameters -------------------------
gen.add("visualization", bool_t, 0, "A boolean parameter for Visualization",
        False)
gen.add("show_image", bool_t, 0, "Show current frame", False)
gen.add("show_background", bool_t, 0, "Show computed background", False)
gen.add("show_diff_image", bool_t, 0,
        "Show moving objects in current frame", False)
gen.add("show_moving_objects_contours", bool_t, 0,
        "Show bounding boxes around moving objects", False)


# ------------------------------- DBSCAN enable -------------------------------
gen.add("dbscan_enable", bool_t, 0, "Enable DBSCAN", True)

exit(gen.generate(PACKAGE, "motion_node", "motion_cfg"))
